### PR TITLE
add doc about effects for capabilities

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

- Explain the effect of a capability in docstr.
- Move tests about capabilities.
- Use correct path for import.
- Use pull_request as the trigger of workflow.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/contrib/capabilities/context_handling.py [hunk 1]
```diff
@@ -25,10 +25,9 @@ class TransformChatHistory:
     2. Second, it limits the number of message to keep
     3. Third, it limits the total number of tokens in the chat history
 
-    Args:
-        max_tokens_per_message (Optional[int]): Maximum number of tokens to keep in each message.
-        max_messages (Optional[int]): Maximum number of messages to keep in the context.
-        max_tokens (Optional[int]): Maximum number of tokens to keep in the context.
+    When adding this capability to an agent, the following are modified:
+    - The `process_all_messages_before_reply` hook is registered to transform the chat history.
```
Comment (ekzhu) [inline autogen/agentchat/contrib/capabilities/context_handling.py:29]:
```suggestion
    - A hook is added to the hookable method `process_all_messages_before_reply` to transform the chat history.
```

File: autogen/agentchat/contrib/capabilities/teachability.py [hunk 2]
```diff
@@ -23,6 +23,12 @@ class Teachability(AgentCapability):
     To make any conversable agent teachable, instantiate both the agent and the Teachability class,
     then pass the agent to teachability.add_to_agent(agent).
     Note that teachable agents in a group chat must be given unique path_to_db_dir values.
+
+    When adding Teachability to an agent, the following are modified:
+    - The agent's system message is appended with a note about the agent's new ability.
+    - The agent's `process_last_received_message` hook is registered for processing the last received message.
```
Comment (ekzhu) [inline autogen/agentchat/contrib/capabilities/teachability.py:29]:
```suggestion
    - A hook is added to the agent's `process_last_received_message` hookable method, and the hook potentially modifies the last of the received messages to include earlier teachings related to the message.
```

**Review by ekzhu — CHANGES_REQUESTED** [review]
We should also emphasize that the capabilities themselves do not modify the stored chat history of the agent, rather they always transform the messages received inside the call to `generate_reply` and `a_generate_reply`, which is called after the latest recieved message is stored.

**Comment by sonichi** [comment]
@ekzhu PTAL

File: autogen/agentchat/contrib/capabilities/teachability.py [hunk 3]
```diff
@@ -23,6 +23,12 @@ class Teachability(AgentCapability):
     To make any conversable agent teachable, instantiate both the agent and the Teachability class,
     then pass the agent to teachability.add_to_agent(agent).
     Note that teachable agents in a group chat must be given unique path_to_db_dir values.
+
+    When adding Teachability to an agent, the following are modified:
+    - The agent's system message is appended with a note about the agent's new ability.
+    - A hook is added to the agent's `process_last_received_message` hookable method,
+    and the hook potentially modifies the last of the received messages to include earlier teachings related to the message.
+    Not modifying the stored message history. Stores apparent teachings in new memos.
```
Comment (rickyloynd-microsoft) [inline autogen/agentchat/contrib/capabilities/teachability.py:31]: "Added teachings do not propagate into the stored message history. If new user teachings are detected, they are added to new memos in the vector database."

**Review by rickyloynd-microsoft — APPROVED** [review]
Nice improvements.

File: autogen/agentchat/contrib/capabilities/teachability.py (hunk 3 above)
Comment (sonichi) [reply autogen/agentchat/contrib/capabilities/teachability.py]: Thanks. Modified.

**Review by ekzhu — APPROVED** [review]

**Comment by sonichi** [comment]
@LittleLittleCloud I changed the trigger back to pull_request to address a security requirement.

**Comment by LittleLittleCloud** [comment]
@sonichi

OK sounds good,  that might prevent PR from forked repo to consume openai secrets?

**Comment by sonichi** [comment]
> @sonichi
> 
> OK sounds good, that might prevent PR from forked repo to consume openai secrets?

Yes. So if a PR requires an openai test, it needs to be made from a branch within this repo.

**Pull request merged**